### PR TITLE
Feat: Added Python Decouple to Configure Environment Variables

### LINKED ISSUE #580: add python-decouple for handling the environment variables
## Description
As a member,
I need to add python-decouple to manage the **DJANGO SECRET KEY** and other sensitive env variables,
so that I can easily set up the project.

### Update [Required]
- [ ] settings/base.py
- [ ] settings/dev.py
- [ ] requirements/prod.txt
 
## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
1 hour

### PR DESCRIPTION
**Pull request by saideepesh000**
### Description
Added Python-decouple to configure environment files

Fixes #580 

### Type of Change:

- Code

**Code/Quality Assurance Only**
- This change requires a documentation update (software upgrade on readme file)

### Checklist:

**Code/Quality Assurance Only**
- [ ] My changes generate no new warnings

### CODE REVIEW

**Review by theyashshahs — CHANGES_REQUESTED** [review]
where is the .env.example file

**Comment by theyashshahs** [comment]
@saideepesh000 you are not fixing anything, it is a new feature so change the PR title accordingly from [commit style guidelines](https://github.com/anitab-org/portal/wiki/Commit-Message-Style-Guide)

File: .gitignore
Comment (SanketDG) [inline .gitignore]:
```suggestion
# Environment variables
```

File: systers_portal/systers_portal/settings/dev.py
Comment (SanketDG) [inline systers_portal/systers_portal/settings/dev.py]: Let's keep the original values as default values here for the sake of sanity when there is no `.env` file.

**Review by SanketDG — CHANGES_REQUESTED** [review]

**Comment by saideepesh000** [comment]
> 
> 
> where is the .env.example file

Actually I don't about this, I searched it online now, now should I delete this PR and submit a new one or this PR can be resubmittable.

**Comment by SanketDG** [comment]
@saideepesh000 You have to just work on the same branch. You can choose to add more commits or just amend the current commit. Feel free to ping on Zulip if you have issues.

**Comment by theyashshahs** [comment]
@saideepesh000 .env.example is an example file that reflects the environment variables of the .env file but not the original ones. 
You can take help from [here](https://simpleisbetterthancomplex.com/2015/11/26/package-of-the-week-python-decouple.html)
and do the necessary changes in the .travis.yml

File: systers_portal/systers_portal/settings/dev.py
Comment (saideepesh000) [reply systers_portal/systers_portal/settings/dev.py]: You want me to remove the config() and keep it as it was in dev.py for all the variables (NAME, USER, PASSWORD, HOST, PORT, EMAIL_HOST, EMAIL_PORT) ?

Comment (SanketDG) [reply systers_portal/systers_portal/settings/dev.py]: No, there is a way to specify default values
```
config('EMAIL_PORT', default=25, cast=int)
```

Just keep the original values as default values.

Comment (saideepesh000) [reply systers_portal/systers_portal/settings/dev.py]: Oh, OK Ok

**Comment by theyashshahs** [comment]
@saideepesh000 do the changes in the .travis.yml :)

**Review by theyashshahs — COMMENTED** [review]
can you trigger a commit to test the TRAVIS CI?

**Comment by theyashshahs** [comment]
the TRAVIS CI test is failing the build is failing because it cannot find the .env.example file

**Comment by saideepesh000** [comment]
> 
> 
> the TRAVIS CI test is failing the build is failing because it cannot find the .env.example file

Now, what should I do? please help me.

**Comment by theyashshahs** [comment]
give a proper path to the .env.example file like in which it is in

**Comment by saideepesh000** [comment]
@yashshah2820  Still, it's failing

**Comment by theyashshahs** [comment]
> @yashshah2820 Still, it's failing

@saideepesh000  If you are still facing problem then you can post the error on the zulip channel of AnitaB.org

**Comment by saideepesh000** [comment]
> 
> 
> @yashshah2820 Still, it's failing

@yashshah2820 @SanketDG What might be the reason For failing of checks, if I run the server I could able to get the website running.

**Comment by SanketDG** [comment]
@saideepesh000 Run `flake8` on the project.

**Comment by saideepesh000** [comment]
> 
> 
> @saideepesh000 Run `flake8` on the project.

Ya I fixed those indendation things, should I resubmit again to this branch.

**Comment by theyashshahs** [comment]
@saideepesh000 yes commit the new changes made 👍

**Comment by saideepesh000** [comment]
@yashshah2820  @SanketDG, Finally checks completed :)

**Review by theyashshahs — DISMISSED** [review]
nice work :)

File: systers_portal/.env.example
Comment (SanketDG) [inline systers_portal/.env.example]: nit: space after key

File: systers_portal/systers_portal/settings/dev.py
Comment (SanketDG) [inline systers_portal/systers_portal/settings/dev.py]: no need to have defaults for user and pass :)

File: systers_portal/.env.example
Comment (SanketDG) [inline systers_portal/.env.example]: nit: no need to add spaces for this, just bunch the above four all of them up together.

**Review by SanketDG — CHANGES_REQUESTED** [review]
This is great @saideepesh000, just a couple of small things left!

Also, don't forget to follow https://github.com/anitab-org/portal/wiki/Commit-Message-Style-Guide while writing the commit message

**Review by SanketDG — DISMISSED** [review]
Great work @saideepesh000, this looks good to merge!

Feel free to [squash](https://github.com/wprig/wprig/wiki/How-to-squash-commits) these commits (you will get to learn squashing 🙂 ) or the person who merges will do it themselves.

**Review by sakshi1499 — DISMISSED** [review]
@saideepesh000 Please squash the commits and follow the commit guidelines for the final commit. It is ready to merge then.

**Review by theyashshahs — DISMISSED** [review]

**Comment by sakshi1499** [comment]
@saideepesh000 Please resolve the conflicts.

**Comment by saideepesh000** [comment]
@sakshi1499 @SanketDG , Could anyone please tell me what's wrong here

**Comment by SanketDG** [comment]
> Feel free to [squash](https://github.com/wprig/wprig/wiki/How-to-squash-commits) these commits (you will get to learn squashing :slightly_smiling_face:) or the person who merges will do it themselves.

You have to squash the commits @saideepesh000

**Comment by saideepesh000** [comment]
> 
> 
> > Feel free to [squash](https://github.com/wprig/wprig/wiki/How-to-squash-commits) these commits (you will get to learn squashing 🙂) or the person who merges will do it themselves.
> 
> You have to squash the commits @saideepesh000

can I use  --force after the squashing to push to this branch, or it may cause any errors?.

**Comment by SanketDG** [comment]
> can I use --force after the squashing to push to this branch, or it may cause any errors?.

Sure! You always need to force push for making changes to an existing commit!

**Comment by saideepesh000** [comment]
@SanketDG, @sakshi1499 , I am confused, can I delete this PR/branch and resubmit it with only one commit.

**Comment by SanketDG** [comment]
@saideepesh000 Yes, you totally can, but you would lose out the chance to learn some Git! You can always ask on Zulip if you are having issues with `git`, otherwise feel free to open a new PR.

**Comment by saideepesh000** [comment]
@SanketDG , should I delete the branch or PR

**Comment by theyashshahs** [comment]
> @SanketDG , should I delete the branch or PR

force push the commits

**Comment by saideepesh000** [comment]
@SanketDG, @yashshah2820 I can't work on this PR anymore, I messed up with my local repo, I am not able to run the project locally, I think because of the upgrades of versions, I saw that merger or maintainer of the repo could squash the commits as well, I feel that the changes were correct but I can't able to squash.

**Comment by SanketDG** [comment]
@saideepesh000 Messing up is totally fine, that is not an issue. Please don't think of it as a bad thing, most people mess up with git when starting out. Yes, I can totally fix this myself and merge this, but please let me know if you are willing to work on this, we could have a call then, and we could work on it together and merge this.

I understand if you do not want to continue as this has taken much of your time, I will fix the PR as soon as I get some time, meanwhile you can choose some other issue to work on, let me know!

**Comment by saideepesh000** [comment]
@SanketDG, Ya, I would love to contribute to this community, I already knew JavaScript, React, for the sake of this project I started learning Django. As soon as I become comfortable with it, I would work on more issues. Meanwhile, I am working with react and JS projects here.

**Review by ritwickraj78 — DISMISSED** [review]
LGTM

**Review by sakshi1499 — DISMISSED** [review]
LGTM

**Comment by SanketDG** [comment]
@saideepesh000 I have mostly fixed out all of the issues, it would be really great if you could just update the README.md with instructions for setting this up like [mentorship-backend](https://github.com/anitab-org/mentorship-backend/#run-app) has! Please let me know, and I will open a issue for you, and you can then create a new PR for that.

**Comment by saideepesh000** [comment]
> @saideepesh000 I have mostly fixed out all of the issues, it would be really great if you could just update the README.md with instructions for setting this up like [mentorship-backend](https://github.com/anitab-org/mentorship-backend/#run-app) has! Please let me know, and I will open a issue for you, and you can then create a new PR for that.

@SanketDG Yes, I would work on it.

**Comment by SanketDG** [comment]
I have created https://github.com/anitab-org/portal/issues/642 for you @saideepesh000 

I will merge this PR in a bit.

**Comment by SanketDG** [comment]
@saideepesh000 Since all dependent work on this PR is completed, feel free to rebase this PR and I will merge this, thanks for all the work!

**Comment by saideepesh000** [comment]
> 
> 
> @saideepesh000 Since all dependent work on this PR is completed, feel free to rebase this PR and I will merge this, thanks for all the work!

@SanketDG, you mean, to resolve the merge conflicts I need to rebase it, right.

**Comment by SanketDG** [comment]
@saideepesh000 Yes.

**Comment by saideepesh000** [comment]
> 
> 
> @saideepesh000 Yes.

@SanketDG, Can I manually do this, because I am afraid of it doing in this project. I may do something wrong or mess up with this.

**Comment by SanketDG** [comment]
How do you mean when you say manually?

**Comment by saideepesh000** [comment]
> 
> 
> How do you mean when you say manually?

@SanketDG, by clicking on "resolve conflicts" button and checking the files ".gitignore", ".travis.yml", "settings/base.py", and editing those file, I think, or is there any other way to do this, please let me know

**Comment by SanketDG** [comment]
Feel free to do that, if it works for you.

**Comment by saideepesh000** [comment]
@SanketDG, actually what I did was, I removed lines which are already present in the develop branch, I'm not sure if this is correct.

**Review by SanketDG — DISMISSED** [review]

**Review by SanketDG — APPROVED** [review]
Finally we merge @saideepesh000 :tada:

**Comment by saideepesh000** [comment]
> 
> 
> Finally we merge @saideepesh000 🎉

@SanketDG, Thanks for helping me throughout this, I got very afraid & tensed when I messed up the whole thing, thanks for your kindness & patience.

**Pull request merged**
